server/swap/swap.go

```diff
@@ -799,7 +799,7 @@ func (s *Swapper) Run(ctx context.Context) {
 						return
 					}
 					select {
-					case <-mainLoop:
+					case <-ctxHelpers.Done():
```

Comment:
This seems unreachable since `mainLoop` is not closed until after all the helpers have returned

Reply:
Looks right. `mainLoop` is intended to be closing the, uh, main loop, not this helper that must die *first*.